### PR TITLE
The Video Endpoint

### PR DESCRIPTION
**Pull request by Dala0**
## What does this change?

Adds the video endpoint.

## How to test

Run 'make test'.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,7 +35,13 @@ clean:
 	make -C referenceapi/ clean
 	make -C genericoptions/ clean
 	make -C test-against-captureddata/ clean
+	make -C video/ clean
+	make -C video/ clean
```
Comment (fredex42) [inline Makefile:39]: don't need it twice!

File: video/video.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	if errResponse != nil {
+		switch errResponse.StatusCode {
+		case 404:
+			return common.MakeResponseRaw(404, aws.String("No content found.\n"), "text/plain"), nil
+		default:
+			return errResponse, nil
+		}
+	}
+
+	if _, ok := (event.QueryStringParameters)["poster"]; ok {
```
Comment (fredex42) [inline video/video.go:35]: I'd say `if _, havePoster` not `if _, ok` as the latter is harder to read

File: common/responses.go [hunk 3]
```diff
@@ -86,3 +96,21 @@ func MakeResponseRaw(responseCode int, contentBodyString *string, contentType st
...
+*/
+func MakeResponseRedirect(uRL string) *events.APIGatewayProxyResponse {
+	headers := getDefaultHeaders()
+	headers["Location"] = fmt.Sprintf("%s", uRL)
+
+	return &events.APIGatewayProxyResponse{
+		StatusCode: 302,
+		Headers:    headers,
+	}
+}
```
Comment (fredex42) [inline common/responses.go:116]: Can we get a quick test for this pls?

**Review by fredex42 — CHANGES_REQUESTED** [review]
Mostly there, just a couple of tweaks

File: Makefile (hunk 1 above)
Comment (Dala0) [reply Makefile:39]: Fixed.

File: video/video.go (hunk 2 above)
Comment (Dala0) [reply video/video.go:35]: Done.

File: common/responses.go (hunk 3 above)
Comment (Dala0) [reply common/responses.go:116]: Done.

**Pull request merged**